tests/unit_tests/test_elasticsearchstore.py

```diff
@@ -301,3 +255,64 @@ def test__update_latest_metric_if_necessary(test_elastic_metric, test_elastic_la
+                         [({'type': 'parameter', 'key': 'param0',
+                            'comparator': 'LIKE', 'value': '%va%'},
+                           Q("term", params__key="param0") &
+                           Q('bool', must=[Q("wildcard", params__value="*va*")]),
+                           "params"),
+                          ({'type': 'parameter', 'key': 'param0',
+                            'comparator': 'ILIKE', 'value': '%va%'},
+                           Q("term", params__key="param0") &
+                           Q('bool', must=[Q("wildcard", params__value="*va*")]),
+                           "params"),
+                          ({'type': 'parameter', 'key': 'param0',
+                            'comparator': '=', 'value': 'va'},
+                           Q("term", params__key="param0") &
+                           Q('bool', must=[Q("term", params__value="va")]),
+                           "params"),
+                          ({'type': 'metric', 'key': 'metric0', 'comparator': '>', 'value': '1'},
+                           Q("term", latest_metrics__key="metric0") &
+                           Q('bool', must=[Q("range", latest_metrics__value={'gt': "1"})]),
+                           "latest_metrics"),
+                          ({'type': 'metric', 'key': 'metric0', 'comparator': '>=', 'value': '1'},
+                           Q("term", latest_metrics__key="metric0") &
+                           Q('bool', must=[Q("range", latest_metrics__value={'gte': "1"})]),
+                           "latest_metrics"),
+                          ({'type': 'metric', 'key': 'metric0', 'comparator': '<', 'value': '1'},
+                           Q("term", latest_metrics__key="metric0") &
+                           Q('bool', must=[Q("range", latest_metrics__value={'lt': "1"})]),
+                           "latest_metrics"),
+                          ({'type': 'metric', 'key': 'metric0', 'comparator': '<=', 'value': '1'},
+                           Q("term", latest_metrics__key="metric0") &
+                           Q('bool', must=[Q("range", latest_metrics__value={'lte': "1"})]),
+                           "latest_metrics"),
+                          ({'type': 'tag', 'key': 'tag0', 'comparator': '!=', 'value': 'val2'},
+                           Q("term", tags__key="tag0") &
+                           Q('bool', must_not=[Q("term", tags__value="val2")]),
+                           "tags")])
+@mock.patch('elasticsearch_dsl.Search.query')
+@pytest.mark.usefixtures('create_store')
+def test___build_elasticsearch_query(search_query_mock, test_parsed_filter, test_query,
+                                     test_type, create_store):
+    search_query_mock.return_value = Search()
+    create_store._build_elasticsearch_query(parsed_filters=[test_parsed_filter], s=Search())
+    search_query_mock.assert_called_with('nested', path=test_type, query=test_query)
```

Comment:
instead of checking assert_called_with, is it possible to check that the query returned by the function is the one you expect ?

Reply:
Which function ? The search_query_mock function doesn't return a query but a Search() object. This a function frol elasticsearch-dsl it should return what I expect right ?

Reply:
If I give it the right input

Reply:
Ok I just understood what you meant